ui/views.py

```diff
@@ -172,7 +172,9 @@ def get_vocabularies(facets):
         vocab = (vocabulary_id, vocabs[int(vocabulary_id)])
         vocabularies[vocab] = []
         for t_id, count in term_data:
```

Comment:
@amir-qayyum-khan @noisecapella 

It _sounds_ like a race condition. The `Term` is deleted, but Elasticsearch isn't yet updated. That suggests one of two things:

1. Simple race condition.
2. Deleting a term doesn't trigger index updates for the `LearningResource`s. 

If it's the first, waiting a few seconds and trying again should "fix" it. If it's the second, then some work would need to be done to fix it. However, that work should be put into the new `elasticsearch-dsl` code, and not here.